azerothcore/azerothcore-wotlk
# fix(DB): Teldrassil herbalism gameobject re-construction

All objects are pulled from sniff data.
Behavior observed while gathering data used to form pools.



## Changes Proposed:
Adds a significant amount of missing herbalism gameobjects for Teldrassil.
Introduces area based pooling to prevent herb extinction.
![image](https://user-images.githubusercontent.com/91289495/149443864-da024956-087a-453e-a7b2-193e18009396.png)
![image](https://user-images.githubusercontent.com/91289495/149443874-6cf02f8f-c1e9-4c41-bb85-4d1f05b1e14a.png)


Overall herbs increased based on count (fully spawned zone)



## Issues Addressed:

- Removes old incorrect pooling.

## SOURCE:

Teldrassil zone packets, tediously gathered

## Tests Performed:

- Every single node inserted has been checked as part of the process I did to pool them


## How to Test the Changes:


1.  Load the SQL and pick a pool you'd like to check.
2.  Check the spawns in that pool and make sure only the number assigned are active.

Note:  It looks a bit heavily spawned due to view distance defaults on AC; turn down view distance in config and it will look more natural, the actual spawns are fine and working properly

## Known Issues and TODO List:

There are likely less than 5, possibly less than 3 missing nodes (possibly none, but I don't give myself that much credit and I feel like I was missing some when I went to check, despite having gathered the packets) in the entire zone now, and they spawn as much as they should on a blizzard server.  There may be minor inaccuracies in the pooling but it is closer than it has ever been to correct.



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
